app/helpers/application_helper.rb

```diff
@@ -42,31 +32,26 @@ def current_journey
     parent
   end
 
-  def user_header_link
-    return admin_header_link if current_journey == :admin
+  def user_header_navigation(header)
+    return if current_journey == :citizens
+    return admin_header_navigation(header) if current_journey == :admin
 
-    provider_header_link
+    provider_header_navigation(header)
   end
 
-  def provider_header_link
+  def provider_header_navigation(header)
     if provider_signed_in?
-      render "shared/provider_header_link"
+      header.with_navigation_item(text: current_provider.username, href: providers_provider_path, active: false)
+      header.with_navigation_item(text: t("layouts.logout.provider"), href: destroy_provider_session_path, active: false, options: { method: :delete }, html_attributes: { method: :delete })
```

Comment:
`html_attributes: { method: :delete }`, this attribute is added to the `<li>` and may not required

Reply:
@naseberry without this I get a Page not found error when I click on the Logout link. I think because by default the link generates a GET action and saml_sessions_controller.rb cannot direct this to the right action (as it requires a DELETE action).

Reply:
👍 thanks @naseberry have removed html_attributes: { method: :delete }